main.js

```diff
@@ -0,0 +1,80 @@
+const Apify = require('apify');
+const { findGlassdoorLocation } = require('./src/find-location');
+const { searchJobs } = require('./src/search-jobs');
+const { parseJobs } = require('./src/parse-jobs');
+const { BASE_URL } = require('./src/consts');
+
+const { log } = Apify.utils;
+
+Apify.main(async () => {
+    log.info('INPUT Validation...');
+    const input = await Apify.getInput();
+    // INPUTS
+    const {
+        proxy,
+        query,
+        maxResults,
+        location,
+        locationstate,
+        // 4 options available to search with: Jobs, Companies, Salaries, Interviews
+        // only Jobs and Companies are used because other two do not fit data model.
+        // THIS ACTOR HAS ONLY 'JOBS' MODE (DECIDED TO MAKE SEPARATE ACTOR FOR 'COMPANY' MODE IF NEEDED)
+        // category = 'Jobs',
+    } = input;
+    const proxyConfiguration = await Apify.createProxyConfiguration(proxy);
+    // CHECKS
+    if (Apify.isAtHome() && !proxyConfiguration) {
+        throw new Error('WRONG INPUT: This actor must use Apify proxy or custom proxies when running on Apify platform!');
+    }
+    if (typeof query !== 'string') {
+        throw new Error('WRONG INPUT: must contain `query` field as string');
+    }
+
+    const proxyUrl = proxyConfiguration ? proxyConfiguration.newUrl() : undefined;
```

Comment:
this can go inside `handleRequestFunction`, using `session.id` as parameter, unless there's no room for "session pool".